tests/neural_networks/test_qcnn.py

```diff
@@ -140,3 +149,33 @@ def test_sequence(self, num_qubits, operations):
                 or mock_quantum_pooling_layer3.call_count > 0
                 or mock_fully_connected_layer.call_count > 0
             )
+
+    @pytest.mark.parametrize(
+        "num_qubits, operations",
+        [
+            (2, [(dict, {})]),
+            (3, [(str, {})]),
+            (4, [(object, {})]),
+            (2, [(MERA, {})]),
+            (3, [(QCNN, {})]),
+        ],
+    )
+    def test_non_baselayer_operation(self, num_qubits, operations):
+        """Tests that classes not inheriting from BaseLayer are rejected."""
+        with raises(
+            TypeError,
+            match=r"Operation at index \d+\ must inherit from BaseLayer, got \*?",
+        ):
+            _ = QCNN(num_qubits).sequence(operations)
+
+    @pytest.mark.parametrize(
+        "num_qubits, operations",
+        [(2, [(BaseLayer, {})])],
+    )
+    def test_baselayer(self, num_qubits, operations):
+        """Tests that BaseLayer itself is rejected."""
+        with raises(
+            TypeError,
+            match=r"Operation at index \d+\ cannot be BaseLayer itself, got \*?",
```

Comment:
Making corresponding changes here.
```suggestion
            match=r"Operation at index \d+\ cannot be BaseLayer itself.",
```